### PR TITLE
TST: Switch to pytest

### PR DESCRIPTION
**Pull request by ericdill**

### CODE REVIEW

File: test_history.py [hunk 1]
```diff
@@ -101,20 +105,20 @@ def test_len():
...
+    assert b, 'aardvark'
 
 
 def test_protected_key():
-    assert_raises(ValueError, h.__getitem__,
-                  HistoryDict.RESERVED_KEY_KEY)
-    assert_raises(ValueError, h.__setitem__,
-                  HistoryDict.RESERVED_KEY_KEY, 'aardvark')
+    with pytest.raises(ValueError):
+        h.__getitem__(HistoryDict.RESERVED_KEY_KEY)
```
Comment (tacaswell) [inline test_history.py]: This can be simplified to `h[HistoryDict.RESERVED_KEY_KEY]`

**Comment by tacaswell** [comment]
I think you can remove the `nose` installation too?

**Comment by ericdill** [comment]
~~Is `nose` still being installed somewhere?~~  I had not rebased on top of upstream/master and so did not see the nose installation in .travis.yml

**Comment by ericdill** [comment]
@tacaswell I think all of the comments are addressed

**Pull request merged**